handler.go

```diff
@@ -15,90 +15,66 @@
 package otel // import "go.opentelemetry.io/otel"
 
 import (
-	"log"
-	"os"
-	"sync"
-	"sync/atomic"
+	"go.opentelemetry.io/otel/internal/debug"
 )
 
 var (
 	// globalErrorHandler provides an ErrorHandler that can be used
 	// throughout an OpenTelemetry instrumented project. When a user
 	// specified ErrorHandler is registered (`SetErrorHandler`) all calls to
 	// `Handle` and will be delegated to the registered ErrorHandler.
-	globalErrorHandler = defaultErrorHandler()
-
-	// delegateErrorHandlerOnce ensures that a user provided ErrorHandler is
-	// only ever registered once.
-	delegateErrorHandlerOnce sync.Once
+	globalErrorHandler = &errorHandlerDelegate{
```

Comment:
Can we remove the changes to the ErrorHandler for this PR?

I'm not entirely convinced this is a good idea (what does it mean to set a logger versus an error handler), but I also don't think my uncertainty should block the progress of adding internal logging. If we can isolate the changes being made in this PR to just adding internal logging it would be easier to get this merged IMO.

Reply:
I also get that it might not be correct to abandon this unification without it being recognized. If that is the case can we create an issue to track the work instead?

Reply:
On second thought the changes to how error handling is delegated are a separate issue.  I originally included it because it was the exact same thing I was going to use to delegate the logger before I realized I don't needed.  This commit will pull those out, and I will open a separate PR.

We should still use the internal logging as the default, this is a much smaller change.  The reasonfor it is what would I expect from changing the logging of the SDK.  I would think if I changed it to a pretty formatter (or json or anything really), I would expect that ALL logs, including those from the contrib libraries usingthe error handler, would start to use the new formater.  I would be surprised, in a bad way, to find out that to get the contrib errors to be formated I would also have to write an error handler.

The behavior that I would expect:
The default is internal logger, so if you change the logger everything from SDK/API and contrib gets the new formatting
The Error handler would be more specific, so if you change that it won't affect the more verbose logs, but it would change Errors from the SDK and contrib.

Reply:
Resolving per https://github.com/open-telemetry/opentelemetry-go/issues/2417